src/pages/Settings/Settings.js

```diff
@@ -1,146 +1,182 @@
-      onChange={handleToggleSshAccess} />
+export function SettingsTable ({ updateAvailable, header, children }) {
+  return <table styleName='settings-table' data-testid='settings-table'>
+    <thead>
+      <tr key='heading'>
+        <th id={header.toLowerCase().trim()} styleName='settings-row-header'>
+          <h5 styleName='row-header-title'>{header}
+            { updateAvailable
+              ? <span styleName='second-header'> Update Available</span>
+              : null
+            }
+          </h5>
+        </th>
+      </tr>
+    </thead>
+    <tbody>
+      {children}
+    </tbody>
+  </table>
+}
 
-    <Button name='factory-reset' variant='danger' wide onClick={() => push('/factory-reset')}>Factory Reset</Button>
-  </PrimaryLayout>
+export function SettingsRow ({ label, content, showSoftwareUpdateModal, updateAvailable, versionTable }) {
+  return <tr styleName='settings-row' data-testid='settings-row'>
+    <td styleName='settings-col align-left'>
+      <h3 styleName='row-label' data-testid='settings-label'>{label}</h3>
+    </td>
+    <td styleName='settings-col align-right'>
+      { updateAvailable && versionTable
+        ? <SoftwareUpdateButton availableVersion={content} showSoftwareUpdateModal={showSoftwareUpdateModal} />
+        : versionTable
+          ? <h3 styleName='row-content'>Your Software is up-to-date</h3>
+          : <h3 styleName='row-content'>{content}</h3>
+      }
+    </td>
+  </tr>
 }
 
-export function SettingsFormInput ({
-  name,
-  label,
-  type = 'number',
-  register,
-  errors = {},
-  ...inputProps
-}) {
-  return <>
-    {label && <label styleName='settingsLabel' htmlFor={name}>{label}</label>}
-    <Input name={name} id={name} type={type} placeholder={label} ref={register} {...inputProps} />
-    {errors[name] && <small styleName='field-error'>{errors[name].message}</small>}
-  </>
+function SoftwareUpdateButton ({ availableVersion, showSoftwareUpdateModal }) {
+  return <Button
+    onClick={() => showSoftwareUpdateModal(availableVersion)}
+    styleName='update-version-button'>
+    Update Software
+  </Button>
 }
 
-const mockedProps = {
-  settings: {
-    hostName: 'My Host',
-    hostUrl: 'https://288f092.holo.host',
-    hostPubKey: 'hcsFAkeHashSTring2443223ee',
-    registrationEmail: 'iamahost@hosting.com',
-    deviceName: 'My Very First HoloPort',
-    networkId: 'my-holoport',
-    sshAccess: false,
-    deviceAdminPort: 6609,
-    hcAdminPort: 8800,
-    hcNetworkPort: 35353,
-    hostingPort: 8080
-  },
-  updateSettings: () => Promise.resolve(true),
-  factoryReset: () => Promise.resolve(true),
-  toggleSshAccess: () => Promise.resolve(true)
+export function UpdateSoftwareModal ({ availableVersion, updateVersion, handleClose, settings }) {
+  if (!availableVersion) return null
+  const onYes = () => {
+    updateVersion(availableVersion)
+    handleClose()
+  }
+  return <Modal
+    contentLabel={`Update ${settings.deviceName}?`}
+    isOpen={!!availableVersion}
```

Comment:
I think overloading `availableVersion` like this makes the code harder to understand. Is it really saving us much to avoid just having a separate `modalIsOpen` state in the parent?